### ISSUE TITLE
Issue with .load_model/1 matching on :zip.unzip (bad_central_directory)

### ISSUE THREAD

**Issue by sitch** [elixir-nx/bumblebee#142]
On arch linux calling:
```elixir
Bumblebee.load_model({:hf, "stanford-crfm/pubmedgpt"})
```
Throws
```elixir
** (MatchError) no match of right hand side value: {:error, :bad_central_directory}                                              
    (bumblebee 0.1.2) lib/bumblebee/conversion/pytorch/loader.ex:29: Bumblebee.Conversion.PyTorch.Loader.load_zip!/1
    (bumblebee 0.1.2) lib/bumblebee/conversion/pytorch.ex:24: Bumblebee.Conversion.PyTorch.load_params!/4
    (bumblebee 0.1.2) lib/bumblebee.ex:399: Bumblebee.load_params/4
    (bumblebee 0.1.2) lib/bumblebee.ex:378: Bumblebee.load_model/2
```
The problem is with <https://github.com/elixir-nx/bumblebee/blob/main/lib/bumblebee/conversion/pytorch/loader.ex#L29>

Seems to be some sort of erlang decoding issue in: <https://github.com/erlang/otp/blob/master/lib/stdlib/src/zip.erl>

The file isn't corrupt, as I am able to unzip using the linux `unzip` binary.  Also, the file size is 10GB, and I currently have >50GB free ram

**Comment by sitch** [comment]
Just tried unzipping and zipping in the cache dir, and getting the same error

**Comment by josevalim** [comment]
zip in Erlang does not support zip64, so it is limited on 2GB. We should probably detect such cases and fallback to `unzip`.

**Comment by sitch** [comment]
Ah! Good catch.

 I'm more than happy to contribute a patch.  
 
 I found some libs supporting zip64.  Let me know which you prefer.

- <https://github.com/toadmaninteractive/eunzip>
- <https://github.com/akash-akya/unzip>
- <https://github.com/evadne/packmatic>

**Comment by jonatanklosko** [comment]
@josevalim the file from [here](https://huggingface.co/CompVis/stable-diffusion-v1-4/tree/main/unet) unzips correctly and it's 3.44 GB, or is the 2 GB limit on a single file in the archive?

**Comment by sitch** [comment]
PR here: https://github.com/elixir-nx/bumblebee/pull/144

**Comment by jonatanklosko** [comment]
Closed in #144.
